### PR TITLE
Four more lifecycle events

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
New Events:-

* `OutOfMemoryCallback` - Runs when the client catches an OutOfMemoryError
* `GameOptionsSavedCallback` - Runs when the game options have been printed to the file
* `LightningStruckCallback` - Runs as soon as `LightningBoltEntity`'s constructor is invoked
* `PlayerDisconnectCallback` - Runs when the player is disconnected from a server

Bumped Fabric api version to 0.2.1 and Lifecycle events v0 version to 0.1.2

### CODE REVIEW

File: fabric-events-lifecycle-v0/build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 archivesBaseName = "fabric-resource-loader-v0"
-version = getSubprojectVersion(project, "0.1.1")
+version = getSubprojectVersion(project, "0.1.2")
```
Comment (hYdos) [inline fabric-events-lifecycle-v0/build.gradle:2]: why did you bump the resource loader version?

File: fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/MixinLightningBoltEntity.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.entity.LightningBoltEntity;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.event.world.LightningStruckCallback;
+
+@Mixin(LightningBoltEntity.class)
+public class MixinLightningBoltEntity {
+	@Inject(at = @At("HEAD"), method = "<init>")
```
Comment (hYdos) [inline fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/event/lifecycle/MixinLightningBoltEntity.java:31]: HEAD init will crash mixin. dont do it use TAIL or RETURN

File: fabric-events-lifecycle-v0/build.gradle (hunk 1 above)
Comment (BoogieMonster1O1) [reply fabric-events-lifecycle-v0/build.gradle:2]: uh oh

Comment (BoogieMonster1O1) [reply fabric-events-lifecycle-v0/build.gradle:2]: lemme do a real quick fix

Comment (BoogieMonster1O1) [reply fabric-events-lifecycle-v0/build.gradle:2]: that's the wrong archivesbasename lemme fix

**Pull request merged**